datadog/dogstatsd/base.py

```diff
@@ -62,6 +73,20 @@ def __init__(self, host='localhost', port=8125, max_buffer_size=50, namespace=No
 
         self.lock = Lock()
 
+        # Check host and port env vars
+        DD_AGENT_HOST = os.environ.get('DD_AGENT_HOST', '')
+        if DD_AGENT_HOST != '' and host == 'localhost':
+            host = DD_AGENT_HOST
+
+        DD_DOGSTATSD_PORT = os.environ.get('DD_DOGSTATSD_PORT', '')
+        if DD_DOGSTATSD_PORT != '' and port == 8125:
+            try:
+                port = int(DD_DOGSTATSD_PORT)
+            except ValueError:
+                log.warning("Port number provided in DD_DOGSTATSD_PORT env var is not an integer: \
+                %s, using %s as port number", DD_DOGSTATSD_PORT, port)
+                pass
+
```

Comment:
We may have to set the default arguments to `None` to detect actual args. @yannmh wdyt?

Reply:
so should I set the default arguments to `None` ? @ofek @yannmh

Reply:
I don't have a strong opinion on this. If we decide to keep the default args, I'd name the default argument, so it becomes
```python
        # Check host and port env vars
        DD_AGENT_HOST = os.environ.get('DD_AGENT_HOST', '')
        if DD_AGENT_HOST and host == DEFAULT_HOST:
            host = DD_AGENT_HOST
```

Other notes:
1. is it a "conventional" to name Python variables that store env variable with UPPERCASE? 
2 `os.environ.get('DD_AGENT_HOST', '')` -> `os.environ.get('DD_AGENT_HOST')`